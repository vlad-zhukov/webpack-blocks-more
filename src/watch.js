/*
 * Webpack file watch.
 */
export default function watch(options) {
    return (context, {merge}) =>
        merge({
            watch: true,
            watchOptions: options,
        });
}
