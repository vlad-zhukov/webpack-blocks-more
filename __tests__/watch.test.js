import {createConfig} from '@webpack-blocks/webpack';
import watch from '../src/watch';

it('should add a watch setting to true', () => {
    const config = createConfig([watch()]);

    expect(config).toMatchSnapshot();
});

it('should take watchOptions', () => {
    const config = createConfig([
        watch({
            aggregateTimeout: 300,
        }),
    ]);

    expect(config).toMatchSnapshot();
});
