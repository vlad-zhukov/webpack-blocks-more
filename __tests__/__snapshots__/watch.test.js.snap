// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a watch setting to true 1`] = `
Object {
  "module": Object {
    "rules": Array [],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
  "watch": true,
  "watchOptions": undefined,
}
`;

exports[`should take watchOptions 1`] = `
Object {
  "module": Object {
    "rules": Array [],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
  "watch": true,
  "watchOptions": Object {
    "aggregateTimeout": 300,
  },
}
`;
